apollo-compiler/src/main/kotlin/com/apollographql/android/compiler/OperationTypeSpecBuilder.kt

```diff
@@ -46,7 +46,7 @@ class OperationTypeSpecBuilder(
 
   private fun TypeSpec.Builder.addQueryDocumentDefinition(fragments: List<Fragment>): TypeSpec.Builder {
     val initializeCodeBuilder = CodeBlock.builder().add(OPERATION_DEFINITION_FIELD_NAME)
-    fragments.forEach {
+    fragments.filter { operation.fragmentsReferenced.contains(it.fragmentName) }.forEach {
```

Comment:
@sav007 fyi, another bug was because we were attempting to append the fragment definition for an operation that doesn't reference that fragment. (See the file below)

Fragments are defined per schema file so here we need to filter out those that aren't referenced in the operation before appending their definition.